### ISSUE TITLE
Install is broken due to missing requests

### ISSUE THREAD

**Issue by WoLpH** [nickovs/unificontrol#3]
The `setup.py` includes `unificontrol` which (after a few more imports) also imports `requests`.

I'm guessing this could be solved by specifying `requests` as `setup_requirements` instead of `install_requirements` but not entirely sure.

```python
Obtaining unificontrol
    ERROR: Command errored out with exit status 1:
     command: unificontrol/.venv/bin/python3.8 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'unificontrol/setup.py'"'"'; __file__='"'"'unificontrol/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info
         cwd: unificontrol/
    Complete output (9 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "unificontrol/setup.py", line 4, in <module>
        import unificontrol
      File "unificontrol/unificontrol/__init__.py", line 30, in <module>
        from .unifi import UnifiClient, FETCH_CERT
      File "unificontrol/unificontrol/unifi.py", line 31, in <module>
        import requests
    ModuleNotFoundError: No module named 'requests'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

To reproduce:
```bash
git clone https://github.com/nickovs/unificontrol.git
cd unificontrol
pipenv install -e .
```

**Comment by nickovs** [comment]
That is indeed a problem. Thanks for catching that. The *only* reason that `setup.py` imports `unificontrol` is to get the version number, so rather than making the dependency management more complex I will just fix the way that it tracks the version number.

**Comment by WoLpH** [comment]
Quick question btw, to make usage slightly easier I've made a few dataclasses that I'm using locally. If you're interested I can make a pull request for those. We could even extend it to make `client.name = 'whatever'` automatically update through the API.

Example:
```python
@dataclasses.dataclass                       
class Client:                                
    _id: str                                 
    assoc_time: int                          
    authorized: bool                         
    first_seen: int                          
    ip: str                                  
    is_guest: bool                           
    is_wired: bool                           
    last_seen: int                           
    latest_assoc_time: int                   
    mac: str                                 
    oui: str                                 
    qos_policy_applied: bool                 
    rx_bytes: int                            
    rx_bytes_r: int                          
    rx_packets: int                          
    site_id: str                             
    tx_bytes: int                            
    tx_bytes_r: int                          
    tx_packets: int                          
    tx_retries: int                          
    uptime: int                              
    user_id: str                             
    wifi_tx_attempts: int                    
    _is_guest_by_uap: bool = None            
    _is_guest_by_ugw: bool = None            
    _is_guest_by_usw: bool = None            
    _last_seen_by_uap: int = 0               
    _last_seen_by_ugw: int = 0               
    _last_seen_by_usw: int = 0               
    _uptime_by_uap: int = 0                  
    _uptime_by_ugw: int = 0                  
    _uptime_by_usw: int = 0                  
    anomalies: int = 0                       
    ap_mac: str = ''                         
    bssid: str = ''                          
    bytes_r: int = 0                         
    ccq: int = 0                             
    channel: int = 0                         
    confidence: int = 0                      
    dev_cat: int = 0                         
    dev_family: int = 0                      
    dev_id: int = 0                          
    dev_vendor: int = 0                      
    device_name: str = ''                    
    dhcpend_time: int = 0                    
    essid: str = ''                          
    fingerprint_source: int = 0              
    fixed_ip: str = ''                       
    fw_version: str = ''                     
    gw_mac: str = ''                         
    hostname: str = ''                       
    idletime: int = 0                        
    is_11r: bool = None                      
    name: str = ''                           
    network: str = ''                        
    network_id: str = ''                     
    noise: int = 0                           
    noted: bool = None                       
    os_class: int = 0                        
    os_name: int = 0                         
    powersave_enabled: bool = None           
    priority: int = 0                        
    radio: str = ''                          
    radio_name: str = ''                     
    radio_proto: str = ''                    
    roam_count: int = 0                      
    rssi: int = 0                            
    rx_rate: int = 0                         
    satisfaction: int = 0                    
    score: int = 0                           
    signal: int = 0                          
    sw_depth: int = 0                        
    sw_mac: str = ''                         
    sw_port: int = 0                         
    tx_power: int = 0                        
    tx_rate: int = 0                         
    use_fixedip: bool = None                 
    usergroup_id: str = ''                   
    vlan: int = 0                            
```

Generated by:
```python
 import collections                                          
 keys = collections.defaultdict(list)                        
 counter = 0                                                 
 for data in client.list_clients():                                        
     counter += 1                                            
     data = {k.replace('-', '_'): v for k, v in data.items()}
     for key, value in data.items():                         
         keys[key].append(value)                             
                                                             
 required = dict()                                           
 optional = dict()                                           
 for key, values in sorted(keys.items()):                    
     type_ = type(values[0])                                 
     if type_ is int:                                        
         default = 0                                         
     elif type_ is str:                                      
         default = ''                                        
     elif type_ is bool:                                     
         default = None                                      
     else:                                                   
         raise TypeError()                                   
                                                             
     type_str = f'{type_.__name__}'                          
                                                             
     if len(values) == counter:                              
         required[key] = type_str                            
     else:                                                   
         optional[key] = f'{type_str} = {default!r}'         
                                                             
 for key, type_ in sorted(required.items()):                 
     print(f'{key}: {type_}')                                
                                                             
 for key, type_ in sorted(optional.items()):                 
     print(f'{key}: {type_}')                                
```

**Comment by nickovs** [comment]
Having some data classes for the various sorts of entities managed by Unifi would be helpful. My only concern with your proposal is that the `dataclasses` module was only added in Python 3.7 and I had been trying to keep the code compatible with 3.6. Perhaps we can find a way to have these classes show up if the `dataclasses` module module is present but be absent if running on an older version of Python. If you want to come up with a PR I'd be happy to take a look.

**Comment by WoLpH** [comment]
For Python 3.6 there's a backport available so that's easily fixed: https://pypi.org/project/dataclasses/
The requirement would be:
```
dataclasses; python_version < '3.7'
```

Another question would be how this new API would be exposed. The way I see it there are a few options:
1. Add an option to client so `client.list_clients()` returns this object when the new API is enabled
2. Add a parameter to methods such as `client.list_client()` so it returns the new style
3. Add a new method for this so that `client.clients` would be a collection class that is both iterable and subscriptable using something like `client.clients[mac_address]`.
4. Add a `dict` api to the dataclasses to maintain backwards compatibility

My vote would be option 3, but that's your decision :)

**Comment by nickovs** [comment]
All good ideas. I've opened a new issue #5 to discuss this further so that I can close this issue once I have committed the fix. Let's move the discussion to there.

**Comment by WoLpH** [comment]
I've moved the messages :)

**Comment by nickovs** [comment]
Fixed in version 0.2.6. New version committed to the master branch and pushed to PyPI.
